app/ts/background/metadataUtils.ts

```diff
@@ -139,25 +140,21 @@ export async function identifyAddress(ethereumClientService: EthereumClientServi
 
 export async function getAddressBookEntriesForVisualiser(ethereumClientService: EthereumClientService, visualizerResult: readonly (VisualizerResult | undefined)[], simulationState: SimulationState, userAddressBook: UserAddressBook) : Promise<AddressBookEntry[]> {
 	let addressesToFetchMetadata: bigint[] = []
-	let tokenAddresses: bigint[] = []
 
 	for (const vis of visualizerResult) {
 		if (vis === undefined) continue
 		const ethBalanceAddresses = vis.ethBalanceChanges.map((x) => x.address)
-		const from = vis.tokenResults.map((x) => x.from)
-		const to = vis.tokenResults.map((x) => x.to)
-		tokenAddresses = tokenAddresses.concat(vis.tokenResults.map((x) => x.tokenAddress))
-		addressesToFetchMetadata = addressesToFetchMetadata.concat(ethBalanceAddresses, from, to)
+		const eventArguments = vis.events.map((x) => x.type !== 'NonParsed' ? x.args : []).flat()
+		const addressesInEvents = eventArguments.filter((a): a is { typeValue: { type: 'address', value: EthereumAddress }, paramName: string } => a.typeValue.type === 'address').map((a) => a.typeValue.value)
```

Comment:
Why `x` on the first line and `a` on the second?

Reply:
no particular reason, I added better var names